src/main/java/ru/curs/celesta/transaction/CelestaTransactionAspect.java

```diff
@@ -43,6 +43,10 @@ public Object execEntryPoint(ProceedingJoinPoint joinPoint) throws Throwable {
     }
 
     private Object proceedInTransaction(CallContext c, ProceedingJoinPoint joinPoint) throws Throwable {
+        if (c.getDBPid() != 0) {

```

Comment:
I think this place deserves a following comment for future maintainers:

```java
/* Skip CallContext activation in case it is already active. Non-zero DBPID 
is not a perfect way to distinguish active CallContexts (e. g. a closed 
CallContext will also have a cached non-zero DBPID), but  since we 
do not have public 'isActive' property on CallContext, it's just
a working solution.
*/
```

Reply:
Done.